api/v1alpha1/module_types.go

```diff
@@ -92,42 +91,112 @@ type KernelMapping struct {
 	Regexp string `json:"regexp"`
 }
 
-// ModuleSpec describes how the OOT operator should deploy a Module on those nodes that need it.
-type ModuleSpec struct {
+type ModprobeArgs struct {
+	// Load is an optional list of arguments to be used when loading the kernel module.
+	// +kubebuilder:validation:MinItems=1
+	Load []string `json:"load,omitempty"`
+
+	// Unload is an optional list of arguments to be used when unloading the kernel module.
+	// +kubebuilder:validation:MinItems=1
+	Unload []string `json:"unload,omitempty"`
+}
+
+type ModprobeSpec struct {
+	// ModuleName is the name of the Module to be loaded.
+	ModuleName string `json:"moduleName"`
 
+	// Parameters is an optional list of kernel module parameters to be provided to modprobe.
+	Parameters []string `json:"parameters,omitempty"`
+
+	// DirName is the root directory for modules.
+	// +kubebuilder:default=/opt
+	DirName string `json:"dirName,omitempty"`
+
+	// Args is an optional list of arguments to be passed to modprobe before the name of the kernel module.
 	// +optional
-	Build *Build `json:"build"`
+	Args *ModprobeArgs `json:"args,omitempty"`
 
+	// If RawArgs are specified, they are passed straight to the modprobe binary; all other properties in this
+	// object are ignored.
```

Comment:
```suggestion
	// If RawArgs are specified, they are passed straight to the modprobe binary; all other properties in this
	// object are ignored.
	// Example: `modprobe ${RawArgs}
`
```

I think this example worth more than any verbal explanation (especially for those who aren't familiar with the modprobe command syntax.

Reply:
PTAL.

Reply:
lgtm